volttrontesting/utils/platformwrapper.py

```diff
@@ -565,18 +583,71 @@ def startup_platform(self, vip_address, auth_dict=None,
                          msgdebug=False,
                          setupmode=False,
                          agent_monitor_frequency=600,
-                         timeout=60):
+                         timeout=60,
+                         # Allow the AuthFile to be preauthenticated with keys for service agents.
+                         perform_preauth_service_agents=True):
         # Update OS env to current platform's env so get_home() call will result
         # in correct home director. Without this when more than one test instance are created, get_home()
         # will return home dir of last started platform wrapper instance
         os.environ.update(self.env)
 
+        # Add check and raise error if the platform is already running for this instance.
+        if self.is_running():
+            raise PlatformWrapperError("Already running platform")
+
         self.vip_address = vip_address
         self.mode = mode
         self.volttron_central_address = volttron_central_address
-        self.volttron_central_serverkey =volttron_central_serverkey
+        self.volttron_central_serverkey = volttron_central_serverkey
         self.bind_web_address = bind_web_address
+
+        if perform_preauth_service_agents:
+            authfile = AuthFile()
+
+            # pre-seed all of the volttron process identities before starting the platform
+            for identity in PROCESS_IDENTITIES:
+                capabilities = ['allow_auth_modifications']
```

Comment:
where is this used? I currently see it only in test cases.

Reply:
this is what was there so I added it here..really need to look and make sure we have things that make sense for this

Reply:
so I modified this code to only do it for masterweb....as @sgilbride  points out we need to update it when his pr comes in.